src/rcv/unicore.c

```diff
@@ -0,0 +1,918 @@
+		3072.0,6144.0,0.0
+	};
+	int i;
+	for (i = 0; i < 15; i++) if (ura_eph[i] >= value) break;
+	return i;
+}
+
+/* get observation data index ------------------------------------------------*/
+static int obsindex(obs_t* obs, gtime_t time, int sat)
+{
+	int i, j;
+
+	if (obs->n >= MAXOBS) return -1;
+	for (i = 0; i < obs->n; i++) {
+		if (obs->data[i].sat == sat) return i;
+	}
+	obs->data[i].time = time;
+	obs->data[i].sat = sat;
+	for (j = 0; j < NFREQ + NEXOBS; j++) {
+		obs->data[i].L[j] = obs->data[i].P[j] = 0.0;
+		obs->data[i].D[j] = 0.0;
+		obs->data[i].SNR[j] = obs->data[i].LLI[j] = 0;
+		obs->data[i].code[j] = CODE_NONE;
+	}
+	obs->n++;
+	return i;
+}
+
+/* adjust weekly rollover of GPS time ----------------------------------------*/
+static gtime_t adjweek(gtime_t time, double tow)
+{
+	double tow_p;
+	int week;
+	tow_p = time2gpst(time, &week);
+	if (tow < tow_p - 302400.0) tow += 604800.0;
+	else if (tow > tow_p + 302400.0) tow -= 604800.0;
+	return gpst2time(week, tow);
+}
+
+/* check code priority and return freq-index ---------------------------------*/
+static int checkpri(const char* opt, int sys, int code, int idx)
+{
+	int nex = NEXOBS;
+
+	if (sys == SYS_GPS) {
+		if (strstr(opt, "-GL1L") && idx == 0) return (code == CODE_L1L) ? 0 : -1;
+		if (strstr(opt, "-GL2S") && idx == 1) return (code == CODE_L2X) ? 1 : -1;
+		if (strstr(opt, "-GL2P") && idx == 1) return (code == CODE_L2P) ? 1 : -1;
+		if (code == CODE_L1L) return (nex < 1) ? -1 : NFREQ;
+		if (code == CODE_L2S) return (nex < 2) ? -1 : NFREQ + 1;
+		if (code == CODE_L2P) return (nex < 3) ? -1 : NFREQ + 2;
+	}
+	else if (sys == SYS_GLO) {
+		if (strstr(opt, "-RL2C") && idx == 1) return (code == CODE_L2C) ? 1 : -1;
+		if (code == CODE_L2C) return (nex < 1) ? -1 : NFREQ;
+	}
+	else if (sys == SYS_GAL) {
+		if (strstr(opt, "-EL6B") && idx == 3) return (code == CODE_L6B) ? 3 : -1;
+		if (code == CODE_L6B) return (nex < 2) ? -1 : NFREQ;
+	}
+	else if (sys == SYS_QZS) {
+		if (strstr(opt, "-JL1L") && idx == 0) return (code == CODE_L1L) ? 0 : -1;
+		if (strstr(opt, "-JL1Z") && idx == 0) return (code == CODE_L1Z) ? 0 : -1;
+		if (code == CODE_L1L) return (nex < 1) ? -1 : NFREQ;
+		if (code == CODE_L1Z) return (nex < 2) ? -1 : NFREQ + 1;
+	}
+	else if (sys == SYS_CMP) {
+		if (strstr(opt, "-CL1P") && idx == 0) return (code == CODE_L1P) ? 0 : -1;
+		if (strstr(opt, "-CL7D") && idx == 0) return (code == CODE_L7D) ? 0 : -1;
+		if (code == CODE_L1P) return (nex < 1) ? -1 : NFREQ;
+		if (code == CODE_L7D) return (nex < 2) ? -1 : NFREQ + 1;
+	}
+	return idx < NFREQ ? idx : -1;
+}
+
+/* signal type to obs code ---------------------------------------------------*/
+static int sig2code(int sys, int sigtype)
+{
+	if (sys == SYS_GPS) {
+		switch (sigtype) {
+		case 0: return CODE_L1C;  // L1C/A 
+		case 3: return CODE_L1L;  // L1C/A 
+		case 5: return CODE_L2P;  // L2P
+		case 6: return CODE_L5I;  // L5 Data 
+		case 9: return CODE_L2W;  // L2P(Y),semi-codeless 
```

Comment:
The R1.4 doc has a footnote for 9 "When the value of Bit 26 is 1, the L2P (Y) in Bit 25 is actually the L2C signal." Any idea how to interpret this, and what could this 'L2C signal' be?

Reply:
Taking a guess, say the L2C signal referred to is the L2C(M) as case 17 could have been used for L2C(L), and say the UM980 uses z-tracking for L2P(Y) matching the CODE_L2W use here, then

```
static int sig2code(int sys, int sigtype, int l2c) {
  if (sys == SYS_GPS) {
   switch (sigtype) {
    case 9: return l2c ? CODE_CODE_L2S : CODE_CODE_L2W;  // L2C(M); L2P(Y),semi-codeless z-tracking
 ...
  else if (sys == SYS_QZS) {
    switch (sigtype) {
      case 9: return l2c ? CODE_CODE_L2S : CODE_CODE_L2W;  // L2C(M); L2P(Y),semi-codeless z-tracking

decode_track_stat() {
 ...
 int l2c = (stat >> 28) & 1;
 ... code = sig2code(*sys, sigtype, l2c)

```

Reply:
Fixed and updated. Thanks.